From ba0cfc841347c4e9130c079227b000087c21a4e7 Mon Sep 17 00:00:00 2001
From: kfting <kfting@nuvoton.com>
Date: Wed, 14 Apr 2021 14:39:26 +0800
Subject: [PATCH] kernel: dts: npcm845 dts update

1. disable memory access for FIU0
2. modify the chip select info under FIU0

Signed-off-by: kfting <kfting@nuvoton.com>
---
 .../dts/nuvoton/nuvoton-common-npcm8xx.dtsi   |  5 +-
 .../boot/dts/nuvoton/nuvoton-npcm845-evb.dts  | 83 ++++++++-----------
 2 files changed, 35 insertions(+), 53 deletions(-)

diff --git a/arch/arm64/boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi b/arch/arm64/boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi
index bde304b524f6..d38fdcabd2f1 100644
--- a/arch/arm64/boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi
+++ b/arch/arm64/boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi
@@ -231,9 +231,8 @@
 			compatible = "nuvoton,npcm845-fiu";
 			#address-cells = <1>;
 			#size-cells = <0>;
-			reg = <0x0 0xfb000000 0x0 0x1000>,
-			<0x0 0x80000000 0x0 0x10000000>;
-			reg-names = "control", "memory";
+			reg = <0x0 0xfb000000 0x0 0x1000>;
+			reg-names = "control";
 			clocks = <&clk NPCM8XX_CLK_SPI0>;
 			clock-names = "clk_ahb";
 			status = "disabled";
diff --git a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts
index 8f657526784f..88522e772707 100644
--- a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts
+++ b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts
@@ -122,61 +122,44 @@
 
 		fiu0: spi@fb000000 {
 			status = "okay";
-			spi-nor@0 {
+			spi-nor@1 {
 				compatible = "jedec,spi-nor";
 				#address-cells = <1>;
 				#size-cells = <1>;
 				spi-rx-bus-width = <2>;
-				reg = <0>;
+				reg = <1>;
 				spi-max-frequency = <5000000>;
-				partitions@80000000 {
-					compatible = "fixed-partitions";
-					#address-cells = <1>;
-					#size-cells = <1>;
-					bbuboot1@0 {
-						label = "bb-uboot-1";
-						reg = <0x0000000 0x80000>;
-						read-only;
-						};
-					bbuboot2@80000 {
-						label = "bb-uboot-2";
-						reg = <0x0080000 0x80000>;
-						read-only;
-						};
-					envparam@100000 {
-						label = "env-param";
-						reg = <0x0100000 0x40000>;
-						read-only;
-						};
-					spare@140000 {
-						label = "spare";
-						reg = <0x0140000 0xC0000>;
-						};
-					kernel@200000 {
-						label = "kernel";
-						reg = <0x0200000 0x400000>;
-						};
-					rootfs@600000 {
-						label = "rootfs";
-						reg = <0x0600000 0x700000>;
-						};
-					spare1@D00000 {
-						label = "spare1";
-						reg = <0x0D00000 0x200000>;
-						};
-					spare2@0F00000 {
-						label = "spare2";
-						reg = <0x0F00000 0x200000>;
-						};
-					spare3@1100000 {
-						label = "spare3";
-						reg = <0x1100000 0x200000>;
-						};
-					spare4@1300000 {
-						label = "spare4";
-						reg = <0x1300000 0x0>;
-					};
-				};
+
+      	partitions@88000000 {
+      		compatible = "fixed-partitions";
+      		#address-cells = <1>;
+      		#size-cells = <1>;
+      		bmc@0{
+      			label = "bmc";
+      			reg = <0x000000 0x2000000>;
+      		};
+      		u-boot@0 {
+      			label = "u-boot";
+      			reg = <0x0000000 0xC0000>;
+      			read-only;
+      		};
+      		u-boot-env@100000{
+      			label = "u-boot-env";
+      			reg = <0x00100000 0x40000>;
+      		};
+      		kernel@200000 {
+      			label = "kernel";
+      			reg = <0x0200000 0x800000>;
+      		};
+      		rofs@800000 {
+      			label = "rofs";
+      			reg = <0xa00000 0x1500000>;
+      		};
+      		rwfs@1f00000 {
+      			label = "rwfs";
+      			reg = <0x1f00000 0x100000>;
+      		};
+      	};
 			};
 		};
 
-- 
2.17.1

